### ISSUE TITLE
Implement Loss Function

### ISSUE THREAD

**Issue by stepanyanhayk** [McDermottHealthAI/MedRAP#7]
Marginal Log-Likelihood Loss:
P_ret(doc_k | patient) = exp(score_k) / Σ_{j=1 to K} exp(score_j)
P_pred(y | patient, doc_k) = Softmax(Predict(Fuse(patient, doc_k)))
L_marginal = -log( Σ_{k=1 to K} [ P_ret(doc_k | patient) * P_pred(y | patient, doc_k) ] )
